subset/switches/src/main/java/switchtest/Main.java

```diff
@@ -2,31 +2,28 @@
 
 public class Main {
 
-	  public static void main(String[] args) {
-
-	    String ipAddress = "192.168.1.2";
-	    int interfacePort = 10;
-
-	    char[] command = new char[2];
-	    int count = 0;
-
-	    try {
-	      if (args != null) {
-	        if (args[0] != null) {
-	          ipAddress = args[0];
-	        }
-
-	        if (args[1] != null) {
-	          interfacePort = Integer.parseInt(args[1]);
-	        }
-	      }
-	    } catch (Exception e) {
-	      System.err.println("Exception main parse args:" + e.getMessage());
-	    }
-
-	    SwitchInterrogator switchInterrogator = new SwitchInterrogator(ipAddress, interfacePort);
-
-	    Thread switchInterrogatorThread = new Thread(switchInterrogator);
-	    switchInterrogatorThread.start();
-	  }
-	}
+  public static void main(String[] args) {
+
+    String ipAddress = "192.168.1.2";
+    int interfacePort = 10;
+
+    try {
+      if (args != null) {
+        if (args[0] != null) {
```

Comment:
best to check length: if (args.length >= 1) {
}

args[0] will never be null. But args[0] might throw an array length exception.

Reply:
fixed as suggested